### ISSUE TITLE
When in 'flat' mode "clickoutFiresChange" fires even though it is set to false

### ISSUE THREAD

**Issue by artknight** [bgrins/spectrum#293]
When "flat" mode is set and "showButtons" is set to false, clickoutFiresChange is set to true, even though it is explicitely set to false during init.

Proposed change on line 242

Existing code:
clickoutFiresChange = !opts.showButtons || opts.clickoutFiresChange,

New code:
clickoutFiresChange = (!opts.showButtons && opts.clickoutFiresChange) ? true : opts.clickoutFiresChange,

**Comment by bgrins** [comment]
Testing this on flat mode, I think "clickout" doesn't have much meaning in a flat picker since it is always opened.

In fact, I think I would *always* want flat pickers to fire change on clickout..  Otherwise clicking the document will cause the value to revert (which is a problem right now on http://bgrins.github.io/spectrum/#modes-flat if you change the value and then click anywhere else on the document).

So to me it looks like this should actually be:

     clickoutFiresChange = flat || !opts.showButtons || opts.clickoutFiresChange

Is there a use case when you want a clickout to revert your changes in a flat mode picker?

**Comment by artknight** [comment]
yeah, I have another object that allows for  background to be uploaded, and so when clicked it fires a picker click... which is not needed since a different element was clicked

**Comment by bgrins** [comment]
Are you listening to the `change` event or the `move` event for the flat picker?  I think what we actually need to do for flat pickers is to not be binding to the clickout event at all since it doesn't really have any meaning.

In that case we *could* trigger a `change` on `move` instead, but then it confuses things with the 'choose' and 'cancel' buttons.  But those buttons don't really make that much sense either, so maybe those should be off by default for flat mode.

**Comment by artknight** [comment]
well, I have the palette and the picker open, so I am listening to both... change for the palette and dragstop for the picker

**Comment by bgrins** [comment]
So **if flat mode**, we could:

1) Trigger a change event when a move happens
2) Set showButtons to false by default
3) Don't bind to the clickout event at all

I think that would solve the issue and make more sense for flat pickers than what currently happens.
